addons/mail/static/src/core/common/message.js

```diff
@@ -107,11 +110,11 @@ export class Message extends Component {
         /** @type {ShadowRoot} */
         this.shadowRoot;
         this.root = useRef("root");
-        onWillStart(() => this.props.registerMessageRef?.(this.props.message, this.root));
         onWillUpdateProps((nextProps) => {
             this.props.registerMessageRef?.(this.props.message, null);
-            this.props.registerMessageRef?.(nextProps.message, this.root);
         });
+        onMounted(() => this.props.registerMessageRef?.(this.props.message, this.root));
+        onPatched(() => this.props.registerMessageRef?.(this.props.message, this.root));
```

Comment:
`onMounted` + `onPatched` = `onRendered`

Reply:
It's actually different, `onRendered` is sync with state change, not yet added to DOM.

I think `useEffect` (no dependency and no cleanup function is the equivalent.